src/app/components/person/manage/manage-person.service.ts

```diff
@@ -0,0 +1,139 @@
+import { Injectable } from '@angular/core';
+import { Contato, ReponseMessage } from 'src/app/Entities';
+import { PeopleApiService } from 'src/app/services/people-api.service';
+import { isNullOrUndefined } from 'util';
+import { FormGroup } from '@angular/forms';
+import { map } from 'rxjs/operators';
+import { Observable } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ManagePersonService {
+  contatos = [];
+  response: ReponseMessage = { message: '', status: '' };
+
+  constructor(private api: PeopleApiService) {}
+
+  appAddContatoComponent(inputContato: Contato) {
+    this.contatos.push(inputContato);
+  }
+
+  removeContato(contato: Contato) {
+    if (contato.id !== 0) {
+      if (!this.dialogConfirmDeleteContaot(contato)) {
+        return;
+      }
+      this.eventRemoveContatoFromDatabase(contato);
+    } else {
+      this.eventRemoveVisualContatoComponent(contato);
+    }
+  }
+
+  private dialogConfirmDeleteContaot(contato: Contato) {
+    if (!confirm(`Você tem certeza que deseja remover ${contato.name}`)) {
+      return true;
+    }
+    return false;
+  }
+
+  eventRemoveContatoFromDatabase(contato: Contato) {
+    this.removeContatoFromDatabase(contato).subscribe();
+  }
+  removeContatoFromDatabase(contato: Contato): Observable<any> {
+    return this.api.deleteContatoById(contato.id).pipe(
+      map(data => {
+        if (data.status === 200) {
+          this.eventRemoveVisualContatoComponent(contato);
+          this.setResponse({
+            status: 'success',
+            message: 'Contato Removido com sucesso'
+          });
+        } else {
+          this.setResponse({
+            status: 'warning',
+            message:
+              'Não foi possível remover contato, Por favor tente novamente mais tarde'
+          });
+        }
+      })
+    );
+  }
+
+  private eventRemoveVisualContatoComponent(contato: Contato) {
+    this.contatos = this.contatos.filter((value, index, arr) => {
+      return value !== contato;
+    });
+  }
+
+  setResponse(event: { status: string; message: string }) {
+    this.response.message = event.message;
+    this.response.status = event.status;
+  }
+
+  formatDate(dateToFormt: string) {
+    const dateArray = dateToFormt.substr(0, 10).split('-');
+    const day = dateArray[2];
+    const month = dateArray[1];
+    const year = dateArray[0];
+    return `${day}/${month}/${year}`;
+  }
+
+  toDatabaseDate(dateStr): string {
+    const [day, month, year] = dateStr.toString().split('/');
+    return `${year}-${month}-${day}`;
+  }
+
+  validateDate(date: string): boolean {
+    const dateRegex = /^([0-2][0-9]|(3)[0-1])(\/)(((0)[0-9])|((1)[0-2]))(\/)\d{4}$/;
+    if (!dateRegex.test(date)) {
+      return false;
+    }
+    return true;
+  }
+
+  isValidFormToSubmit(fg: FormGroup) {
+    if (
+      this.ifFormGroupInvalid(fg) ||
```

Comment:
Todas essas validações podem ser tiradas daqui se você usar o ReactiveFormsModule

Reply:
ReactiveForms in place.
No more manual validations